File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
  let answer = 0;

  const boardObj = {};

  for (let i = 0; i < board.length; i++) {
    for (let j = 0; j < board.length; j++) {
      boardObj[j + 1]
        ? boardObj[j + 1].push(board[i][j])
        : (boardObj[j + 1] = [board[i][j]]);
    }
  }

  // moves 배열에 맞춰 번호 담기
  const removeArray = [];
  for (const num of moves) {
    for (let i = 0; i < board.length; i++) {
      if (boardObj[num][i] !== 0) {
        removeArray.push(boardObj[num][i]);
        boardObj[num][i] = 0;
        break;
      }
    }
  }

  // 붙어있는 번호 제거하기
  const stack = [];
  for (const item of removeArray) {
    if (!stack.length) {
      stack.push(item);
    } else {
      if (stack[stack.length - 1] === item) {
        stack.pop();
        answer += 2;
      } else {
        stack.push(item);
      }
    }
  }

  return answer;
}

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def removeItem(array, item):
    n = len(array)
    for i in range(n):
            if array[i][item - 1] != 0:
                removedItem = array[i][item - 1]
                array[i][item - 1] = 0
                return [array, removedItem]
    return [array, False]

def solution(board, moves):
    answer = 0
    
    stack = []
    
    for target in moves:
        [board, result] = removeItem(board, target)
        if result != False:
            if not len(stack):
                stack.append(result)
            else:
                if stack[-1] == result:
                    answer += 2
                    stack.pop()
                else:
                    stack.append(result)
            
    return answer